### ISSUE TITLE
openstack reboot: virtual login-node hostnames change

### ISSUE THREAD

**Issue by themisa** [clustervision/trinity#334]
* r5
f178005

When we reboot the openstack node (preceded by nova stop login-a , + however many virtual login nodes are running),

then log back in, nova start login-a, 
sinfo:

```
root@login-a ~]# sinfo
slurm_load_partitions: Unable to contact slurm controller (connect failure)
[root@login-a ~]# hostname
login-a.novalocal
```

**Comment by abhimukh** [comment]
hostname needs to be added to `/etc/hostname`

**Comment by abhimukh** [comment]
It looks like the hostname has already been added to `/etc/hostname` in the branch r5_fixes

```
[root@controller trinity]# cat /trinity/version
* r5_fixes
59ad57c
[root@controller trinity]# grep -A 2 -B 2 hostname login/setup.sh 

#--------------------------------------------------------------------------
# Set hostname and domainname 
#--------------------------------------------------------------------------
echo “login” >> /etc/hostname
hostname login.vc-a
#domainname vc-a

```

Assigning it to @themisa for testing.

**Comment by abhimukh** [comment]
This is a surprisingly difficult problem to solve. So looks like simply changing `/etc/hostname` is not sufficient. This file gets recreated every time the VM is rebooted, which resets its hostname. For r5, I suggest the manual workaround: after rebooting the VM
```
[root@openstack ~(keystone_a)]# ssh login-a
root@login-a's password: 
Last login: Wed Jul 29 08:57:00 2015 from login-a.vc-a
[root@login-a ~]# hostname login
```

**Comment by themisa** [comment]
* r5
34a01f3
This allows slurm to restart on login node.

However:
```
root@node001 ~]# docker-enter trinity
[root@c001 /]# sinfo
sinfo: error: Unable to resolve "login": Host name lookup failure
sinfo: error: Unable to establish control machine address
slurm_load_partitions: Resource temporarily unavailable
[root@c001 /]# cat /etc/resolv.conf 
nameserver 10.141.255.254
search vc-a
```

Do we need to fix something in containers as well?

**Comment by hansthen** [comment]
You need to make sure the login node is added to DNS as well. This needs to be done via xCAT now. 

As a workaround I suggest you do a 

nodeadd login-a groups=login
Then add a login entry to the hosts table

However, this should be done automatically when the login node is booted.

**Comment by abhimukh** [comment]
The default route on the openstack node is wrong
```
[root@openstack ~]# route 
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
default         10.1.101.101    0.0.0.0         UG    0      0        0 eno4
10.1.0.0        0.0.0.0         255.255.0.0     U     0      0        0 eno4
10.141.0.0      0.0.0.0         255.255.0.0     U     0      0        0 eno1
link-local      0.0.0.0         255.255.0.0     U     1002   0        0 eno1
link-local      0.0.0.0         255.255.0.0     U     1005   0        0 eno4
192.168.32.0    0.0.0.0         255.255.252.0   U     0      0        0 br100
```

**Comment by abhimukh** [comment]
The default route is set to go through the company gateway. If we want that to happen then we need to add a route for the 172.16.0.0/12 subnet.

**Comment by hansthen** [comment]
Fixed

**Comment by hansthen** [comment]
Closed as duplicate.

**Comment by abhimukh** [comment]
Fixed in r5_fixes : 68957b396f7e0d79decad166e08510e0f0b53504

I created a systemd unit file to change the hostname after cloud-init has finished running.

**Comment by themisa** [comment]
The node shut down and when i repowered, the hostname was still set to login. i did have to nova-start node again.
The hostname problem however seems fixed.
```
[root@openstack ~(keystone_a)]# nova list
+--------------------------------------+---------+---------+------------+-------------+------------------------------------------+
| ID                                   | Name    | Status  | Task State | Power State | Networks                                 |
+--------------------------------------+---------+---------+------------+-------------+------------------------------------------+
| d41c864f-c60b-4283-b6bc-323f7aa44a17 | login-a | SHUTOFF | -          | Shutdown    | novanetwork=192.168.32.2, 172.16.255.254 |
+--------------------------------------+---------+---------+------------+-------------+------------------------------------------+
[root@openstack ~(keystone_a)]# nova start login-a
[root@openstack ~(keystone_a)]# nova list
+--------------------------------------+---------+--------+------------+-------------+------------------------------------------+
| ID                                   | Name    | Status | Task State | Power State | Networks                                 |
+--------------------------------------+---------+--------+------------+-------------+------------------------------------------+
| d41c864f-c60b-4283-b6bc-323f7aa44a17 | login-a | ACTIVE | -          | Running     | novanetwork=192.168.32.2, 172.16.255.254 |
+--------------------------------------+---------+--------+------------+-------------+------------------------------------------+
[root@openstack ~(keystone_a)]# ssh login.vc-a
Warning: Permanently added 'login.vc-a,172.16.255.254' (ECDSA) to the list of known hosts.
root@login.vc-a's password: 
sed: read error on /usr/share/Modules/init/.modulespath: Is a directory
[root@login ~]# hostname
login
```
